sdk/core/switcher/entry.S

```diff
@@ -339,6 +402,15 @@ exception_entry_asm:
 	reloadRegisters c1, cgp, c4, c5, c6, c7, c8, c9, c10, c11, c12, c13, c14, c15, csp
 	mret
 
+.Lreentrant_exception:
```

Comment:
I'm not sure what we're expecting to happen here.  The exception code runs with interrupts disabled, so we should only ever hit this path if we take a trap while handling an interrupt.  That can only happen in the case of total internal consistency failure and at that point we should just die.  We could achieve that by setting the exception entry point to null on exception entry and then have the hardware assert reset if we take an exception with no exception handler registered.